src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.HttpContentAsyncStream.cs

```diff
@@ -142,6 +149,15 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
                 }
             }
 
+            public override void Write(byte[] buffer, int offset, int count)
+            {
+                // WriteAsync should be how writes are performed on this stream as
+                // part of HttpContent.CopyToAsync.  However, we implement Write just
+                // in case someone does need to do a synchronous write or has existing
+                // code that does so.
+                WriteAsync(buffer, offset, count, CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
Could this deadlock if there's a single threaded SynchronizationContext?  There's a nice article about it [here](http://blogs.msdn.com/b/pfxteam/archive/2011/01/13/10115163.aspx). :stuck_out_tongue: